json-rpc/tests/integration_test.rs

```diff
@@ -858,6 +860,82 @@ fn create_test_cases() -> Vec<Test> {
                 }
             },
         },
+        Test {
+            name: "get_transactions_with_proofs",
+            run: |env: &mut testing::Env| {
+                let resp = env.send("get_metadata", json!([]));
+                assert!(resp.libra_ledger_version > 0);
+
+                // We want to verify the signatures of the LedgerInfo that will be returned by the
+                // get_transactions_with_proofs call to be sure it's valid, but
+                // since we don't have a local state with the set of validators unlike an actual client,
+                // we need to get the validator set using a get_state_proof call.
+                let state_data = env.send("get_state_proof", json!([0])).result.unwrap();
+                let ep_cp = state_data["epoch_change_proof"].as_str().unwrap();
+                let epoch_proofs:EpochChangeProof = lcs::from_bytes(&hex::decode(&ep_cp).unwrap()).unwrap();
+                let some_li:Vec<_> = epoch_proofs.ledger_info_with_sigs;
+                assert!(!some_li.is_empty());
+                // Let's use the first one since the validator set does not change in the tests.
+                let validator_set = &some_li.first().unwrap().ledger_info().next_epoch_state().unwrap().verifier;
+
+                // We test 2 cases:
+                //      1. base_version + limit < resp.libra_ledger_version
+                //      2. base_version + limit > resp.libra_ledger_version
+                for base_version in &[0, resp.libra_ledger_version] {
+                    let limit = 50;
```

Comment:
I'm not sure case 1 is covered here, because I don't know whether server latest ledger version > 50

Reply:
I can add another assert, but [the assert here](https://github.com/libra/libra/pull/6399/files#diff-f1a808a01f2b2b1b2b8b8ab82534890eff0039c5e41cd409ed37bc8d07a1619cR901-R902) ensure we have either 1 or 50 txs in the results, so it works. 
But I can duplicate the test if you prefer.

https://github.com/libra/libra/pull/6399/files#diff-f1a808a01f2b2b1b2b8b8ab82534890eff0039c5e41cd409ed37bc8d07a1619cR901-R902

Reply:
then add an assertion abort the for loop: assert resp.libra_ledger_version > limit
move limit = 50 above of the assert, 

also, I think we can change 50 to smaller number like 5 or 10, in case test is unreliable if limit is too much and server does not generate enough versions.